packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -54,6 +58,10 @@ class TextMapPropagator {
     this._injectB3SingleHeader(spanContext, carrier)
     this._injectTraceparent(spanContext, carrier)
 
+    if (injectCh.hasSubscribers) {
```

Comment:
Why are we applying changes in `TextMapPropagator` and not in `HttpPropagator`? Shouldn't be enough with `HttpPropagator`?

Reply:
They are the same:

`class HttpPropagator extends TextMapPropagator {}`

Reply:
`HttpPropagator` extends `TextMapPropagator`,  so they are not the same. What I want to say is that in the tracer there is defined `HttpPropagator` as `HTTP_HEADERS` and `TextMapPropagator` as `TEXT_MAP`:

```javascript
    this._propagators = {
      [formats.TEXT_MAP]: new TextMapPropagator(config),
      [formats.HTTP_HEADERS]: new HttpPropagator(config),
      [formats.BINARY]: new BinaryPropagator(config),
      [formats.LOG]: new LogPropagator(config)
    }
```

Do we want to apply to both, or only to `HTTP_HEADERS`?

Reply:
both, there are plugins like aws sdk, google cloud pub sub, kafka, etc using `TEXT_MAP` to inject or extract trace data from attributes, metadata or headers.
I guess, the propagation in those cases should behave the same as in an http service